### PR TITLE
Add torch tests to MarginalEstimator and MirrorDescentEstimator

### PR DESCRIPTION
**Pull request by jakee417**
### Your checklist for this pull request
Please review the [guidelines for contributing](CONTRIBUTING.md) to this repository.

- [X] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side). Don't request your master!
- [X] Make sure you are making a pull request against the **dev branch** (left side). Also you should start *your branch* off *our dev*.
- [X] Check the commit's or even all commits' message styles matches our requested structure.

### List of changes to the codebase in this pull request
- Adds tests for the `torch.Tensor` backend.
- Adds a compat_fn for torch/np exp.
- Small typos
- Adds documentation for `FactorDict.from_dict` and `ClusterGraph.clique_beliefs`

### CODE REVIEW

File: pgmpy/estimators/MirrorDescentEstimator.py [hunk 1]
```diff
@@ -45,7 +46,7 @@ def _calibrate(self, theta, n):
         log_z = logsumexp(mu[clique].values)
         for clique in cliques:
             mu[clique] += np.log(n) - log_z
```
Comment (ankurankan) [inline pgmpy/estimators/MirrorDescentEstimator.py:48]: @jakee417 This should be `mu[clique].values` ?

File: pgmpy/estimators/MirrorDescentEstimator.py [hunk 2]
```diff
@@ -45,7 +46,7 @@ def _calibrate(self, theta, n):
         log_z = logsumexp(mu[clique].values)
```
Comment (ankurankan) [inline pgmpy/estimators/MirrorDescentEstimator.py:46]: This would need a compat_fn:
```python
def logsumexp(arr):                                                                                                                                                                                                                                                                                                                                     
     if isinstance(arr, np.ndarray):                                                                                                                                                                                                                                                                                                                     
         return special.logsumexp(arr)                                                                                                                                                                                                                                                                                                                   
     else:                                                                                                                                                                                                                                                                                                                                               
         return torch.logsumexp(arr, 0)   
```

**Review by ankurankan — CHANGES_REQUESTED** [review]
Thanks for the PR. Have added a couple of minor comments.

File: pgmpy/estimators/MirrorDescentEstimator.py (hunk 1 above)
Comment (jakee417) [reply pgmpy/estimators/MirrorDescentEstimator.py:48]: `mu[clique]` is a `DiscreteFactor` which implements addition by a `float`.

File: pgmpy/estimators/MirrorDescentEstimator.py (hunk 2 above)
Comment (jakee417) [reply pgmpy/estimators/MirrorDescentEstimator.py:46]: Similar story here, `logsumexp(mu[clique].values)` reduces to a float which is compatible with `mu[clique] += np.log(n) - log_z`. `scipy.special.logsumexp` is also compatible with `torch.Tensor`:
```python
import torch
from scipy.special import logsumexp

logsumexp(torch.Tensor([10, 10, 10]))
```
When `axis=None`, this reduces to a float.

**Comment by jakee417** [comment]
@ankurankan Added an explicit test for a junction tree with multiple cliques to affirm any doubts about the above comments. Also refactored the torch tests so that they can share code, a similar thing can be applied to `test_MaximumLikelihood.py`

**Review by jakee417 — COMMENTED** [review]
@ankurankan I am not sure how this review feature works, but have addressed your previous comments

**Review by ankurankan — APPROVED** [review]

**Comment by ankurankan** [comment]
@jakee417 Sorry for the delay. I think it looks good. If you are using any references for the implementation could you please add that in References section of the docstring (something like this: https://github.com/pgmpy/pgmpy/blob/cc2260e50712c72d71623474483bef17e15aa4a4/pgmpy/estimators/PC.py#L45)
